code/model/Translatable.php

```diff
@@ -1726,18 +1726,25 @@ public function AllChildrenIncludingDeleted($context = null)
      */
     public function MetaTags(&$tags)
     {
-        $template = '<link rel="alternate" type="text/html" title="%s" hreflang="%s" href="%s" />' . "\n";
-        $translations = $this->owner->getTranslations();
-        if ($translations) {
-            $translations = $translations->toArray();
-            $translations[] = $this->owner;
-            
-            foreach ($translations as $translation) {
-                $tags .= sprintf($template,
-                Convert::raw2xml($translation->Title),
-                i18n::convert_rfc1766($translation->Locale),
-                $translation->AbsoluteLink()
-                );
+        if (
+            ($controller = Controller::curr())
+            && count($controller->getRequest()->getVars())
+        ) {
+            $tags .= '<link rel="canonical" href="'.$controller->AbsoluteLink().'" />';
```

Comment:
This is a preference thing, but maybe you could switch the condition, run this logic then return early, leaving the rest of the logic in this method where it is. What do you think?